game_event/tests.py

```diff
@@ -252,3 +273,48 @@ def test_fail_add_player_to_event_already_in_it(self, client, game_event, player
         assert response.status_code == 200
         with pytest.raises(IntegrityError):
             response = client.post(url, {'answer': 'yes'})
+
+
+@pytest.mark.django_db
+class TestGetWeatherFromApi:
+    def test_one_valid_response(self, game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+        output = get_weather_from_api([game_event], mock_urlopen)
+        game_event.weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
+        assert output == [game_event]
+
+    def test_exception_thrown_from_api_call(self, game_event, player, mock_urlopen):
+
+        mock_urlopen.side_effect = Exception()
+
+        output = get_weather_from_api([game_event], mock_urlopen)
+        assert output[0].weather is None
+
+    def test_one_invalid_response_status_code(self, game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 185
+        mock_urlopen.return_value = mock_response
+
+        output = get_weather_from_api([game_event], mock_urlopen)
+        assert output[0].weather is None
+
+    def test_empty_game_events_list_valid_empty_dictionary_response(self, player):
+        output = get_weather_from_api([], "does not matter")
+        assert output == []
+
+    def test_two_valid_game_events_list_valid_response(self, game_event, another_game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+
+        output = get_weather_from_api([game_event, another_game_event], mock_urlopen)
+        game_event.weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
+        another_game_event.weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
+        assert output == [
+          game_event,
+          another_game_event
+        ]
```

Comment:
Same comment here about the two sides of the equation being references to the same object.